clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -134,11 +147,17 @@ private SslEngineFactory instantiateSslEngineFactory(Map<String, Object> configs
             sslEngineFactory = Utils.newInstance(sslEngineFactoryClass);
         }
         sslEngineFactory.configure(configs);
-        this.sslEngineFactoryConfig = configs;
```

Comment:
the caller will check new ssle engine factory later so we should not update the sslEngineFactoryConfig here